### PR TITLE
Added node list collapse button, restyled collapse

### PR DESCRIPTION
**Pull request by theflyingzamboni**
New on-hover tabs for collapsing/expanding sidebar and node categories. Subject to further changes, but want to go ahead and get the functionality merged.

closes joeyballentine/chaiNNer#122

### CODE REVIEW

File: src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx [hunk 1]
```diff
@@ -102,123 +107,188 @@ const NodeSelector = memo(({ schemata, height }: NodeSelectorProps) => {
         return [...byCategories.values()].flat().filter((n) => favorites.has(n.schemaId));
     }, [byCategories, favorites]);
 
+    const [showCollapseButtons, setShowCollapseButtons] = useState(false);
+
+    const defaultIndex = [['', favoriteNodes]].concat([...byCategories]).map((_, index) => index);
```
Comment (RunDevelopment) [inline src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx:112]: This is an array with the number 0 to `byCategories.size` (inclusive).
```suggestion
    const defaultIndex = Array.from({ length: byCategories + 1 }, (_, i) => i)
```

File: src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx [hunk 2]
```diff
@@ -102,123 +107,188 @@ const NodeSelector = memo(({ schemata, height }: NodeSelectorProps) => {
...
+    const [accordionIndex, setAccordionIndex] = useState<ExpandedIndex>(defaultIndex);
+
+    const toggleAccordion = () => {
+        if (typeof accordionIndex !== 'number' && accordionIndex.length === 0)
+            setAccordionIndex(defaultIndex);
+        else setAccordionIndex([]);
```
Comment (RunDevelopment) [inline src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx:118]: Please use curly braces.

File: src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx [hunk 3]
```diff
@@ -102,123 +107,188 @@ const NodeSelector = memo(({ schemata, height }: NodeSelectorProps) => {
         return [...byCategories.values()].flat().filter((n) => favorites.has(n.schemaId));
     }, [byCategories, favorites]);
 
+    const [showCollapseButtons, setShowCollapseButtons] = useState(false);
+
+    const defaultIndex = [['', favoriteNodes]].concat([...byCategories]).map((_, index) => index);
+    const [accordionIndex, setAccordionIndex] = useState<ExpandedIndex>(defaultIndex);
+
+    const toggleAccordion = () => {
+        if (typeof accordionIndex !== 'number' && accordionIndex.length === 0)
```
Comment (RunDevelopment) [inline src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx:116]: Please extract this condition out into a variable. It's not trivial, and you're using it further below again.

File: src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx (hunk 1 above)
Comment (theflyingzamboni) [reply src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx:112]: This fix throws up an error about trying to add a Map<> to a number. Is this the change you had in mind?
```suggestion
    const defaultIndex = Array.from({ length: [...byCategories].length + 1 }, (_, i) => i);
```

Comment (RunDevelopment) [reply src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx:112]: My bad, it has to be `byCategories.size`. I fixed my suggestion.

**Comment by theflyingzamboni** [comment]
Ignore the PR being closed. I have no idea how it happened, I wasn't even doing any code related at the time it apparently happened.

**Review by RunDevelopment — APPROVED** [review]
Let's get this in for now. We have a bunch of other components with the `useColorModeValue` problem, so what's one more?

**Pull request merged**